sdk/ml/azure-ai-ml/tests/internal/unittests/test_component.py

```diff
@@ -339,12 +340,57 @@ def test_additional_includes(self) -> None:
-            assert ZipFile(code_path / "library.zip").namelist() == ["hello.py", "world.py"]
+            assert ZipFile(code_path / "library.zip").namelist() == ["library/", "library/hello.py", "library/world.py"]
             assert (code_path / "library1" / "hello.py").exists()
             assert (code_path / "library1" / "world.py").exists()
 
         assert not code_path.is_dir()
 
+    def test_additional_includes_ignore(self) -> None:
+        test_configs_dir = Path("./tests/test_configs/internal/")
+        yaml_path = test_configs_dir / "component_with_additional_includes" / "helloworld_additional_includes.yml"
+        additional_includes_dir = test_configs_dir / "additional_includes"
+        component: InternalComponent = load_component(source=yaml_path)
+        # create some files/folders expected to ignore
+        code_pycache = yaml_path.parent / "__pycache__"
+        additional_includes_ignore = additional_includes_dir / "library1" / "x.additional_includes"
+        additional_includes_pycache = additional_includes_dir / "library1" / "__pycache__"
+        try:
+            code_pycache.mkdir()
+            (code_pycache / "a.pyc").touch()
+            additional_includes_ignore.touch()
+            additional_includes_pycache.mkdir()
+            (additional_includes_pycache / "a.pyc").touch()
+            # resolve and check snapshot directory
+            with component._resolve_local_code() as code:
+                code_path = code.path
+                assert not (code_path / "__pycache__").exists()
+                assert not (code_path / "library1" / "x.additional_includes").exists()
+                assert not (code_path / "library1" / "__pycache__").exists()
+        finally:
+            if code_pycache.is_dir():
+                shutil.rmtree(code_pycache)
+            if additional_includes_ignore.is_file():
+                additional_includes_ignore.unlink()
+            if additional_includes_pycache.is_dir():
+                shutil.rmtree(additional_includes_pycache)
+
+    def test_additional_includes_merge_folder(self) -> None:
+        yaml_path = (
+            "./tests/test_configs/internal/component_with_additional_includes/additional_includes_merge_folder.yml"
+        )
+        component: InternalComponent = load_component(source=yaml_path)
+        assert component._validate().passed, repr(component._validate())
+        with component._resolve_local_code() as code:
+            code_path = code.path
+            # first folder
+            assert (code_path / "library1" / "__init__.py").exists()
+            assert (code_path / "library1" / "hello.py").exists()
+            # second folder content
+            assert (code_path / "library1" / "utils").is_dir()
+            assert (code_path / "library1" / "utils" / "__init__.py").exists()
+            assert (code_path / "library1" / "utils" / "salute.py").exists()
+
     @pytest.mark.parametrize(
```

Comment:
Please help to add related test_anonymous_component_reuse. You can merge the tests in my previous commit (as discussed, I'll revert my workaround in my PR) and respect your calculated snapshot id. We need such a test to avoid breaking anonymous reuse

Reply:
OK, will update that test